### PR TITLE
Enabling StyleCop Rules SA1500, SA1511, SA1519

### PR DESCRIPTION
**Pull request by peterfreiling**
Enabling StyleCop Rules:
SA1500 Braces for multi-line statements must not share line
SA1511 While-do footer must not be preceded by blank line
SA1519 Braces must not be omitted from multi-line child statement

### CODE REVIEW

**Comment by cybertyche** [comment]
It's going to take me a little while to get through these. Most of these are clear stylistic improvements. A few of them look worse - the main one so far is the do-while loop having while on its own line.

File: Sources/Core/Microsoft.StreamProcessing/Collections/CircularBuffer.cs [hunk 1]
```diff
@@ -255,8 +255,9 @@ public T PeekLast()
         public IEnumerator<T> GetEnumerator()
         {
             foreach (var buffer in this.buffers)
-                foreach (var item in buffer.Iterate())
-                    yield return item;
+            {
+                foreach (var item in buffer.Iterate()) yield return item;
+            }
```
Comment (cybertyche) [inline Sources/Core/Microsoft.StreamProcessing/Collections/CircularBuffer.cs]: This one makes me a sad panda, but I understand.

File: Sources/Core/Microsoft.StreamProcessing/Collections/FastDictionary.cs [hunk 2]
```diff
@@ -119,7 +120,8 @@ public bool Lookup(TKey key, int hashCode, out int index)
 
                 index++;
                 if (index == this.Size) index = 0;
-            } while (true);
+            }
+            while (true);
```
Comment (cybertyche) [inline Sources/Core/Microsoft.StreamProcessing/Collections/FastDictionary.cs]: Maybe it's the ancient C++ training I had way back in the bronze age, but seeing the while in a do-while on its own line just looks confusing to me. I'll accept it if we need to for consistency with other rules, but this one in particular hurts my brain.

**Comment by peterfreiling** [comment]
Sure no rush. I agree about the do-while, I wish they would have made an exception for it. Initially I thought SA1500 would allow that but that enforces no blank line, not no new line. But since there were only a few in the codebase I thought it was a good tradeoff.

---
In reply to: [493624700](https://github.com/microsoft/Trill/pull/91#issuecomment-493624700) [](ancestors = 493624700)

File: Sources/Core/Microsoft.StreamProcessing/Egress/Temporal/TemporalEgressObservable.cs [hunk 3]
```diff
@@ -41,9 +41,11 @@ public IDisposable Subscribe(IObserver<StreamEvent<TPayload>> observer)
             if (!Config.ForceRowBasedExecution && this.source.Properties.IsColumnar && typeof(TPayload).CanRepresentAsColumnar() && CanGenerateColumnar())
                 pipe = GetPipe(observer);
             else
+            {
                 pipe = new StreamEventEgressPipe<TPayload>(
-                observer,
-                this.container);
+                    observer,
+                    this.container);
+            }
```
Comment (cybertyche) [inline Sources/Core/Microsoft.StreamProcessing/Egress/Temporal/TemporalEgressObservable.cs]: Both the if and else clauses here just assign a value to pipe. Perhaps we should instead convert to a conditional assignment pipe = … ? … : …

File: Sources/Core/Microsoft.StreamProcessing/StringHandling/MultiString.cs [hunk 4]
```diff
@@ -708,9 +708,15 @@ public unsafe ColumnBatch<long> Contains2(string str, ColumnBatch<long> inBV, bo
...
             for (int i = 0; i < this.Count; i++)
+            {
                 if ((bv[i >> 6] & (1L << (i & 0x3f))) == 0)
+                {
                     if (!(largestr.IndexOf(str, startscol[i] + 2, lengthscol[i], StringComparison.Ordinal) >= 0))
+                    {
                         rbv[i >> 6] |= (1L << (i & 0x3f));
+                    }
+                }
+            }
```
Comment (cybertyche) [inline Sources/Core/Microsoft.StreamProcessing/StringHandling/MultiString.cs]: This change hurts my eyes, my heart, and for some reason my spleen. I get it for consistency with the rule, but it really looked good before. :-(

File: Sources/Microsoft.StreamProcessing.ruleset [hunk 5]
```diff
@@ -133,13 +135,11 @@
     <Rule Id="SA1508" Action="None" /> <!-- Closing braces must not be preceded by blank line -->
     <Rule Id="SA1509" Action="None" /> <!-- Opening braces must not be preceded by blank line -->
     <Rule Id="SA1510" Action="None" /> <!-- Chained statement blocks must not be preceded by blank line -->
-    <Rule Id="SA1511" Action="None" /> <!-- While-do footer must not be preceded by blank line -->
     <Rule Id="SA1512" Action="None" /> <!-- Single-line comments must not be followed by blank line -->
     <Rule Id="SA1513" Action="None" /> <!-- Closing brace must be followed by blank line -->
     <Rule Id="SA1514" Action="None" /> <!-- Element documentation header must be preceded by blank line -->
     <Rule Id="SA1515" Action="None" /> <!-- Single-line comment must be preceded by blank line -->
     <Rule Id="SA1516" Action="None" /> <!-- Elements must be separated by blank line -->
```
Comment (cybertyche) [inline Sources/Microsoft.StreamProcessing.ruleset]: A suggestion only for future style refactoring therapy: rules 1512, 1514, and 1515.

**Pull request merged**